### PR TITLE
RET-3725 - Premium for prod only

### PR DESCRIPTION
**Pull request by furnivall**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RET-3725

### Change description ###
parametrised service bus to allow us to only give premium to prod!

**Does this PR introduce a breaking change?** (check one with "x")

```
[x] Yes
[ ] No
```

### CODE REVIEW

File: servicebus-parametrised.tf [hunk 1]
```diff
@@ -1,13 +1,13 @@
-module "servicebus-namespace-premium" {
```
Comment (timja) [inline servicebus-parametrised.tf:1]: you need to use a `moved` block to rename the terraform resource without deleting

File: servicebus-parametrised.tf [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
+  name                = "${var.product}-${var.env}"
   resource_group_name = azurerm_resource_group.rg.name
   location            = var.location
   env                 = var.env
   common_tags         = var.common_tags
-  zone_redundant      = true
-  sku                 = "Premium"
-}
\ No newline at end of file
+  zone_redundant      = var.servicebus_zone_redundant #currently unused, we don't think we need redundancy currently but could change.
```
Comment (timja) [inline servicebus-parametrised.tf:11]: to meet HMCTS resiliency requirements production services should all be ran across multiple available zones

Comment (furnivall) [reply servicebus-parametrised.tf:11]: Switched back to redundant setup

File: servicebus-parametrised.tf (hunk 1 above)
Comment (furnivall) [reply servicebus-parametrised.tf]: think this is sorted now

File: servicebus-parametrised.tf [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  providers = {
+    azurerm.private_endpoint = azurerm.private_endpoint
+  }
+  source              = "git@github.com:hmcts/terraform-module-servicebus-namespace?ref=master"
+  name                = "${var.product}-${var.env}"
+  resource_group_name = azurerm_resource_group.rg.name
+  location            = var.location
+  env                 = var.env
+  common_tags         = var.common_tags
+  zone_redundant      = var.servicebus_zone_redundant #currently unused, we don't think we need redundancy currently but could change.
```
Comment (timja) [inline servicebus-parametrised.tf:11]:
```suggestion
  zone_redundant      = var.servicebus_zone_redundant
```

**Review by timja — APPROVED** [review]

**Review by hsjhita — APPROVED** [review]

**Pull request merged**